packages/template-retail-react-app/app/pages/store-locator/index.jsx

```diff
@@ -5,51 +5,63 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React, {useState} from 'react'
+import React, {useState, useRef} from 'react'
 import {useIntl} from 'react-intl'
-import {Box, Container} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {Box, Container, Button} from '@salesforce/retail-react-app/app/components/shared/ui'
 import Seo from '@salesforce/retail-react-app/app/components/seo'
 import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-content'
 import {useSearchStores} from '@salesforce/commerce-sdk-react'
 import {
-    DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
     DEFAULT_STORE_LOCATOR_POSTAL_CODE,
     STORE_LOCATOR_DISTANCE
 } from '@salesforce/retail-react-app/app/constants'
 import {useForm} from 'react-hook-form'
 
 const StoreLocator = () => {
     const intl = useIntl()
-
+    const [userHasSetManualGeolocation, setUserHasSetManualGeolocation] = useState(false)
     const [searchStoresParams, setSearchStoresParams] = useState({
-        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
-        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE
+        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY.countryCode,
+        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE,
+        limit: 10
     })
     const form = useForm({
         mode: 'onChange',
         reValidateMode: 'onChange',
         defaultValues: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode
+            countryCode: userHasSetManualGeolocation ? searchStoresParams.countryCode : '',
+            postalCode: userHasSetManualGeolocation ? searchStoresParams.postalCode : ''
         }
     })
+    const searchStoresDataRef = useRef({
```

Comment:
why did you choose to use `useRef` instead of `useState`?

Reply:
The `useSearchStores` has to follow the rules of a hook, which prevents it from being called within an event listener, so it has to be at the top level of a component which means it runs at every render. The data that comes out of it had to be in a `useRef` because otherwise using the setter from `useState` would trigger infinite re-renders.

Also, I used `useRef` to make the stores data persistent across re-renders of the component, so that I can run a conditional on `isLoading` to only change the stores data when the data finishes asynchronously loading, otherwise the UI will do some "jumping" between a populated list and a loading state and not flow smoothly from one list to another.

Reply:
try this

```diff
var {data: searchStoresData, isLoading} = useSearchStores({
        parameters: {
            countryCode: searchStoresParams.latitude ? undefined : searchStoresParams.countryCode,
            postalCode: searchStoresParams.latitude ? undefined : searchStoresParams.postalCode,
            latitude: searchStoresParams.countryCode ? undefined : searchStoresParams.latitude,
            longitude: searchStoresParams.countryCode ? undefined : searchStoresParams.longitude,
            locale: intl.locale,
            maxDistance: STORE_LOCATOR_DISTANCE,
            limit: searchStoresParams.limit,
            offset: 0
        }
-    })
+    }, {keepPreviousData: true})
```

Reply:
https://tanstack.com/query/v4/docs/framework/react/reference/useQuery

Reply:
This worked great! Thank you